terraspace/app/modules/lambda/main.tf

```diff
@@ -36,7 +32,8 @@ resource "aws_lambda_function" "lambda_function" {
 }
 
 resource "aws_sns_topic_subscription" "topic_lambda" {
-  topic_arn = data.aws_sns_topic.source_sns_topic.arn
+  for_each  = { for topic in var.sns_topic_trigger_arns : topic => topic }
```

Comment:
Could you check to see what, if any, difference there is with this versus `for_each = toset(var.sns_topic_trigger_arns)`